src/java_bytecode/java_types.cpp

```diff
@@ -233,14 +284,97 @@ typet java_type_from_string(const std::string &src)
   case 'Z': return java_boolean_type();
   case 'V': return java_void_type();
   case 'J': return java_long_type();
-
+  case 'T':
+  {
+    // parse name of type variable
+    INVARIANT(src[src.size()-1]==';', "Generic type name must en on ';'.");
+    irep_idt type_var_name(src.substr(1, src.size()-2));
+    return java_generic_typet(type_var_name);
+  }
   case 'L':
     {
```

Comment:
Consider pulling this case out into its own function

Reply:
I agree with this assessment as a matter of consistency with the other cases.

Reply:
this seems consistent with the other more complex case `'L'`